### PR TITLE
added: filter functionality

### PR DESCRIPTION
**Pull request by rkapoor10**
added filter functionality

- feature - can apply more than one filter 
- all - selects all videos so remove it to see other filters in action
- will work on doing this automatically
- `filterCategories = [ "All", "Training", "Matches", "Exercise", "Doubles", "Singles", "Women", "Men",]`
ss

### CODE REVIEW

**Review by JantuDeb — APPROVED** [review]
The code looks fine. 
There is no issues with the filters implementation.  I would suggest you to use `useReducer()` as you dispatching action and then updating the states in the `filterDispath` function. Basically you created your own version of `useReducer()` hooks. So, if you use reducer the you don't need to update the state by using `setFilter()`, `useReducer()`  will take care of that.

File: src/App.js [hunk 1]
```diff
@@ -25,6 +27,17 @@ function App() {
           <Link to="/auth">Auth ||</Link>
           <Link to="/brokenpath">Broken Link</Link> */}
```
Comment (t007rushi) [inline src/App.js:28]: comments if not needed

File: src/components/cards/VideoCard.jsx [hunk 2]
```diff
@@ -3,20 +3,22 @@ import "./videoCard.css";
...
 import { RiPlayList2Fill } from "react-icons/ri";
 import { Link } from "react-router-dom";
+import {toast} from "react-toastify"
+
 
 const VideoCard = ({details}) => {
   const handleAddToWatchLater = (event) => {
     event.stopPropagation();
+    toast.success("Added to Watch Later")
     console.log("clicked add to watchlater");
```
Comment (t007rushi) [inline src/components/cards/VideoCard.jsx]: console here

File: src/context/FilterContext/FilterContext.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        return setFilters((filtersArray) =>
+          filtersArray.filter((filterItem) => filterItem !== filterCategory)
+        );
+      default:
+        return filtersArray;
+    }
+  };
+
+  const filterData = () => {
+    console.log({ videosData });
```
Comment (t007rushi) [inline src/context/FilterContext/FilterContext.jsx]: same here

File: src/context/FilterContext/FilterContext.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { useVideo } from "../VideoContext/VideoContext";
+
+// create Context
+const defaultContextValue = {};
+const FilterContext = createContext(defaultContextValue);
+
+const FilterProvider = ({ children }) => {
+  const { videosData } = useVideo();
+  const [filtersArray, setFilters] = useState(["All"]);
+  const filterDispatch = (action, filterCategory) => {
```
Comment (t007rushi) [inline src/context/FilterContext/FilterContext.jsx]: separate out reducer, services and contexts

**Review by t007rushi — APPROVED** [review]
check for suggestions

File: src/context/FilterContext/FilterContext.jsx (hunk 4 above)
Comment (rkapoor10) [reply src/context/FilterContext/FilterContext.jsx]: done

File: src/context/FilterContext/FilterContext.jsx (hunk 3 above)
Comment (rkapoor10) [reply src/context/FilterContext/FilterContext.jsx]: done

File: src/components/cards/VideoCard.jsx (hunk 2 above)
Comment (rkapoor10) [reply src/components/cards/VideoCard.jsx]: done

File: src/App.js (hunk 1 above)
Comment (rkapoor10) [reply src/App.js:28]: they are quick nav tools in case side bar breaks while creating functionality...will remove in the final commit

**Comment by rkapoor10** [comment]
> The code looks fine. There is no issues with the filters implementation. I would suggest you to use `useReducer()` as you dispatching action and then updating the states in the `filterDispath` function. Basically you created your own version of `useReducer()` hooks. So, if you use reducer the you don't need to update the state by using `setFilter()`, `useReducer()` will take care of that.

Done

**Pull request merged**